controllers/doguVolumeManager.go

```diff
@@ -264,5 +264,16 @@ func (c *checkIfPVCIsResizedStep) waitForPVCResize(ctx context.Context, doguReso
 }
 
 func isPvcStorageResized(pvc *corev1.PersistentVolumeClaim, quantity resource.Quantity) bool {
+	// first check if the filesystem resize is "pending", which means that the pod has to be (re)started to actually resize the volume
+	// see https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/#file-system-expansion
+	for _, condition := range pvc.Status.Conditions {
```

Comment:
This is delicate stuff because we return on different conditions for different storage controllers. Currently I reflect on this longhorn issue longhorn/longhorn#2749 which states that the matching event `FileSystemResizePending` IS raised but nothing can happen until the mounting pod is up.

We should at least log both occasions to have more insight if things still go wrong but for other reasons.

Another thing. This method is quite comment-y. This is okay as long as code cannot express these circumstances. Yet I feel we can push the envelope a little bit further like with variables but first I think we need to review our procedure of scaling up and down the deployment.

Reply:
Indeed Longhorn has a different status setting than the controller for Azure disks. We keep using this methology because it adds fail-safety for other controllers at best. At minimum this is a longhorn workaround.